tests/ranfunc/gitopsztp/tests/gitopsztp_policies_app.go

```diff
@@ -0,0 +1,128 @@
+		// Check that the required clusters are present
+		err := cluster.CheckClustersPresent(clusterList)
+		if err != nil {
+			Skip(fmt.Sprintf("error occurred validating required clusters are present: %s", err.Error()))
+		}
+
+		// Check for minimum ztp version
+		By("Checking the ZTP version", func() {
+			if !ranfunchelper.IsVersionStringInRange(
+				ranfunchelper.ZtpVersion,
+				"4.10",
+				"",
+			) {
+				Skip(fmt.Sprintf(
+					"unable to run test on ztp version '%s' as it is less than minimum '%s",
+					ranfunchelper.ZtpVersion,
+					"4.10",
+				))
+			}
+		})
+	})
+
+	Context("Apply and Validate custom source CRs on the du policies", Label("ztp-custom-source-crs"), func() {
+
+		var (
+			// customSourceCrPolicyName = "custom-source-cr-policy-config" // enforce policy
+			testCrName = "custom-source-cr"
+			testNs     = "default"
+		)
+
+		It("verifies new CR kind that does not exist in ztp "+
+			"container image can be created via custom source-cr", polarion.ID("61978"), func() {
+			// The ztp test data is stored in a nested directory within the ztp repo
+			testGitPath := gitopsztphelper.JoinGitPaths(
+				[]string{
+					gitopsztphelper.ArgocdApps[gitopsztpparams.ArgocdPoliciesAppName].Path,
+					"ztp-test/custom-source-crs/new-cr",
+				},
+			)
+
+			By("Checking if the git path exists", func() {
+				if !gitopsztphelper.DoesGitPathExist(
+					gitopsztphelper.ArgocdApps[gitopsztpparams.ArgocdPoliciesAppName].Repo,
+					gitopsztphelper.ArgocdApps[gitopsztpparams.ArgocdPoliciesAppName].Branch,
+					testGitPath+"/kustomization.yaml",
+				) {
+					Skip(fmt.Sprintf("git path '%s' could not be found", testGitPath))
+				}
+			})
+
+			By("Checking Service Account does NOT exist on spoke", func() {
+				err := serviceaccount.NewBuilder(ranfuncinittools.SpokeAPIClient, testCrName, testNs).Exists()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(err).To(BeFalse())
+			})
+
+			By("Updating the Argocd app", func() {
+				// Update the Argo app to point to the new test kustomization
+				err := gitopsztphelper.SetGitDetailsInArgocd(
+					gitopsztphelper.ArgocdApps[gitopsztpparams.ArgocdPoliciesAppName].Repo,
+					gitopsztphelper.ArgocdApps[gitopsztpparams.ArgocdPoliciesAppName].Branch,
+					testGitPath,
+					gitopsztpparams.ArgocdPoliciesAppName,
+					true,
+					true,
+				)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			By("Checking new custom Service Account created and applied to spoke", func() {
+				err := serviceaccount.NewBuilder(ranfuncinittools.SpokeAPIClient, testCrName, testNs).Exists()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(err).To(BeTrue())
+			})
+		})
+
+		AfterEach(func() {
+			// Delete leftovers from the custom source-crs test
+			By("Deleting created Service Account from spoke if exists", func() {
+				err := serviceaccount.NewBuilder(ranfuncinittools.SpokeAPIClient, testCrName, testNs).Delete()
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+	})
+
+	AfterEach(func() {
```

Comment:
Why do you need to use 2 AfterEach blocks?

Reply:
the one inside Context labeled with "ztp-custom-source-crs" is intent for my feature tear down after each function.

the one (AfterEach block) outside my own feature context is intent for overall GitOps ZTP policies test suite which can contain/include other policies related feature tests.

Reply:
I understand, 
perhaps it is better to move the second AfterEach (the one that is not labeled) up to be located just below the BeforeEach one? that way we can differentiate between the two different AfterEach.
it is not mandatory, I believe it would be more readable and clear that way. 
WDYT?

Reply:
+1. done.